content_test.go

```diff
@@ -437,3 +437,66 @@ func TestResolve_Repository_WithTargetPlatformOptions(t *testing.T) {
 		t.Fatalf("oras.Resolve() error = %v, wantErr %v", err, errdef.ErrNotFound)
 	}
 }
+
+func TestGenerateDescriptor(t *testing.T) {
+	// create a descriptor successfully
+	desc, err := oras.GenerateDescriptor([]byte("foo"), "example media type")
+	if err != nil {
+		t.Errorf("could not generate descriptor, error = %v", err)
+	}
+	if desc.MediaType != "example media type" {
+		t.Error("wrong media type")
+	}
+	if desc.Digest != digest.FromBytes([]byte("foo")) {
+		t.Error("wrong digest")
+	}
+	if desc.Size != int64(len("foo")) {
+		t.Error("wrong size")
+	}
+
+	// an empty media type will cause GenerateDescriptor to fail
+	_, err = oras.GenerateDescriptor([]byte("bar"), "")
+	if !errors.Is(err, oras.ErrInvalidMediaType) {
+		t.Errorf("unexpected error. expected error = %v, got error = %v", oras.ErrInvalidMediaType, err)
+	}
+}
+
+func TestEqual(t *testing.T) {
```

Comment:
We need test cases for:
1. Same media type, digest and size
2. Different media type
3. Different digest
4. Different size
5. Empty descriptors

Reply:
Clarification needed for item 4: is it different size only? If the descriptors are valid, it's not possible to have different sizes and the same digests. Do we need to test malformed descriptors?

Reply:
Added the rest of tests accordingly.